week1/homework/src/server.js

```diff
@@ -5,16 +5,44 @@ const http = require('http');
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
+function handleRequest(response, state) {
+  response.writeHead(200, { 'content-type': 'application/json' });
+  response.write(JSON.stringify({ state }));
+}
+
+function handleError(response) {
+  response.writeHead(404, { 'content-type': 'application/json' });
+  response.write(JSON.stringify({ error: 'Not found' }));
+}
+
 function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        handleRequest(response, state);
+        break;
+      case '/add':
+        state++;
+        handleRequest(response, state);
+        break;
+      case '/subtract':
+        state--;
+        handleRequest(response, state);
+        break;
+      case '/reset':
+        state = 10;
+        handleRequest(response, state);
+        break;
+      default:
+        handleError(response);
+    }
+    response.end();
```

Comment:
response.end after break 👍

Reply:
Thanks a lot for your feedback! I appreciate it! :)